docs/design/2020-05-08-standardize-error-codes-and-messages.md

```diff
@@ -0,0 +1,222 @@
+# Proposal: Standardize error codes and messages
+
+- Author(s):     Joshua
+- Last updated:  May 8
+- Discussion at: https://docs.google.com/document/d/1beoa5xyuToboSx6e6J02tjLLX5SWzmqvqHuAPEk-I58/edit?usp=sharing
+
+## Abstract
+
+This issue proposes that TiDB components maintain standard error codes and error messages according to a consistent specification.
+
+## Motivation
+
+When certain errors occur in TiDB components, users are often unaware of the meaning of the corresponding error message, and we plan the following two initiatives to alleviate this problem
+
+- Specification of error codes for TiDB components
+- Provide a program that allows users to quickly access information corresponding to error codes
+
+## Proposal
+
+### The Metafiles
+
+In order to let TiUP know the the errors every component may throw, the components developers should
+keep a metafile in the code repository. The metafile can be a json file, a toml file or a markdown file.
+
+#### metafile in JSON
```

Comment:
can we move these rejected formats (JSON and Markdown) into the "Rationale" section? It's very distracting, people reading the RFC from top to bottom will first assume the Metafile format is JSON, and then assume all 3 format can be used, and only after reaching the "Summary" section we know that only TOML is recommended.